### ISSUE TITLE
Better design for anchor links

### ISSUE THREAD

**Issue by matthewp** [stealjs/stealjs#66]
`<h2>` - `<h4>` links appear with an anchor link beside them like in the below screenshot.

![screen shot 2017-08-30 at 5 03 04 pm](https://user-images.githubusercontent.com/361671/29894916-2a22036c-8da5-11e7-8576-ed12b5b5419c.png)

I stole this idea from GitHub which does the same thing:

![screen shot 2017-08-30 at 5 04 30 pm](https://user-images.githubusercontent.com/361671/29894972-5edbc098-8da5-11e7-815d-874cf0590a60.png)

On stealjs.com there is one flaw, it causes the heading to not be lined up with content.

This can be fixed by adding padding to the content and negative margin to the heading like so:

![screen shot 2017-08-30 at 5 06 37 pm](https://user-images.githubusercontent.com/361671/29895039-9755c932-8da5-11e7-8b01-4c4482e669e1.png)

Question: Is this a good solution? Does it look awkward to have that much left padding on the content? Is there a better solution? Am I overthinking this?

**Comment by m-mujica** [comment]
I did not read the whole thing but is this the same issue as https://github.com/stealjs/steal/issues/921 ?

**Comment by andrejewski** [comment]
I'm all for more padding (though it should equal on the right), but maybe instead just put the anchor link to the right of the heading. Also the whole heading should be a click target for the anchoring.

**Comment by matthewp** [comment]
@m-mujica No, that improved it by making them not invisible when not hovered over, but there is still the weirdness with the heading not lining up with the content.
